### DISCUSSION TITLE
Creating a large ".cgns" file

### DISCUSSION THREAD

**Discussion by kondrahin241996** [CGNS/CGNS#371] [CGNStalk]
Hi everybody. I wrote a code with which i can convert files from "_.mesh_" format to "_.cgns_" format.
I have this problem:
I have 2 variants of _cgns-3.2_ and _4.1.2_ library versions.
And when converting a "_.mesh_" file to "_.cgns_" using the cgns 3.2 library, I can see its structure in the _cgnsView_ program. Everything is displayed well.
But when I convert the file using the _cgns 4.1.2_ library, the structure is not displayed in full, not all information is recorded when viewed in _cgnsView_.
What could be the problem?
Maybe the .mesh file is too big to convert? (>50GB).
This is very important to me, please help.

**Comment by kondrahin241996** [comment]
The code is written in C++

**Comment by awoimbee** [comment]
> the structure is not displayed in full, not all information is recorded when viewed in cgnsView.

This is not enough detail to provide an answer.

> I have 2 variants of cgns-3.2 and 4.1.2 library versions.

How ? Are you sure you're not using cgnsview 3.2 on CGNS 4.1 files ?

**Reply by kondrahin241996** [reply]
Good evening.
I'm sorry, I check the file created using the _cgns 4.1.2_ library using the cgnslist utility, not cgnsview.
When viewing a file with the command: _cgnslist -f convertfile.cgns_ only information about the coordinates is displayed.

I open information from the same file that was converted using the _cgns 3.2.1_ library in _cgnsView_ - I have 1 base, 1 zone information about coordinates, and so on. I did everything according to the documentation.
What could be the problem?(

**Comment by kondrahin241996** [comment]
dear users, no one knows what the problem is?(

**Reply by brtnfld** [reply]
Are you sure there was not an error happening in your code when cgns 4.1.2 is used?

Just to take 3.2.1 out of the picture. Are you saying that you convert a .mesh file using your program (which is linked to cgns 4.1.2) and the output from the cgns 4.1.2 cgnslist's output is missing data fields? And using the 4.1.2's cgnsview the data is also missing.

Did you run cgnscheck on the cgnsfile? You could also try dumping the cgns file using HDF5's "h5dump" and verify the data is in the cgns file. (use "h5dump -Hp filename" so you don't print all the raw data). 

Did you try reading the missing data? There might be a bug in cgnslist.

What program generates ".mesh" files? MeshLab?
